src/js/utils/date-utils.js

```diff
@@ -0,0 +1,283 @@
+                determinedZone = 'UTC';
+            }
+            return determinedZone;
+        };
+
+        /**
+         *
+         * Attempt to offset original time based on timezone.
+         *
+         * @param {Object} dateTimeObject A moment.js datetime object.
+         * @param {string} userTZ A timezone representation, e.g. 'America/New_York'.
+         * @return {object} A moment.js datetime object in a determined timezone.
+         */
+        localizeTime = function(dateTimeObject, userTZ) {
+            var localTime;
+            if (isValid(dateTimeObject) && isValid(userTZ)) {
+                localTime = dateTimeObject.tz(userTZ);
+            } else if (isValid(dateTimeObject)) {
+                localTime = dateTimeObject;
+            } else {
+                localTime = '';
+            }
+            return localTime;
+        };
+
+        /**
+         *
+         * Generate a timezone-aware and language-aware string.
+         *
+         * This includes a hook set of built-ins for standard formatting.
+         *
+         * Language is currently overridden by platform-set language preference. If no preference,
+         * the function attempts to determine the locale settings (e.g. en-US) from the browser preferences.
+         *
+         * This function will default to UTC/en-US display in default format in the case of
+         * an unresolvable language or timezone
+         *
+         * preferred_language = 'en' yields Oct. 14, 2016 09:00 BST
+         *
+         * preferred_language = 'ru' yields 14 окт 2016 г. 09:00 BST
+         *
+         * @param {object} localTime A moment.js datetime object offset to a determined timezone.
+         * @param {string} language An ISO 639-1 language string (e.g. 'en', 'fr').
+         * @param {string} format (optional) A desired format constant.
+         * @return {string} A language/timezone aware localized string
+        */
+        timeDisplay = function(localTime, language, format) {
+            var displayLang = 'en-US';
+            var displayFormat = DATE_FORMATS.DEFAULT_DATE_TIME_FORMAT;
+            var displayTime;
+            if (isValid(language)) {
+                displayLang = language;
+            } else {
+                displayLang = window.navigator.userLanguage || window.navigator.language;
+            }
+
+            // Determine the desired format, default to DEFAULT_DATE_TIME_FORMAT
+            if (isValid(format) && format in DATE_FORMATS) {
+                displayFormat = DATE_FORMATS[format];
+            }
+            displayTime = localTime.locale(displayLang).format(displayFormat);
+            // Attempt to repair to UTC/US
+            if (isValid(displayTime) === false) {
+                displayTime = localTime.tz('UTC').locale('en').format(displayFormat);
+            }
+            if (isValid(displayTime) === false) {
+                displayTime = '';
+            }
+            return displayTime;
```

Comment:
Consider splitting up this function into 3 helpers.  So in the end, this function would just be:
```
var displayLang = _getDisplayLang(language)
var displayFormat = _getDisplayFormat(format)
return _getDisplayTime(displayLang, displayFormat)
```